### ISSUE TITLE
ASP.NET to Box

### ISSUE THREAD

**Issue by jetarver** [box/box-windows-sdk-v2#121]
Hi, I'm attempting to build an asp.net 4.0 c# web site that, using strictly server-side c# code, connects/authenticates to our Box.Com account and downloads files using the Box Windows SDK V2 and OAuth2. Sadly, I haven't made it very far.  I have the following BoxConfig and BoxClient objects instantiated:

BoxConfig boxConfig = new BoxConfig(CLIENT_ID, CLIENT_SECRET, new Uri("https://xxxxxxxx.com"));

BoxClient client = new BoxClient(boxConfig);

Can someone please provide me with some code samples on where I go from here?

**Comment by Rod-Sychev** [comment]
Hello,

There good set of samples on github: https://github.com/box/box-windows-sdk-v2

**Comment by jetarver** [comment]
What version of the asp.net framework is required for Box SDK V2?  I’m using 4.0, but can not get an Await or Async to work.

Jason Tarver, MCTS, Security+ | Leidos
Web Applications Developer | Engineering
NEW mobile: 865.255.3025
jason.e.tarver@leidos.com<mailto:jason.e.tarver@leidos.com> | leidos.com/engineering

**Comment by brianytang** [comment]
http://stackoverflow.com/questions/19421878/how-can-i-use-the-async-keywords-in-a-project-targeting-net-4-0/19421907#19421907

If you're referencing the package from nuget directly -- it should download the appropriate libraries for you automatically (eg. Microsoft.Bcl.Async)

**Comment by jetarver** [comment]
Thanks for the replies. I pulled in both the box SDK v2 and Microsoft Async from nuget. Afterwards, I'm able to see that Await is now part of intellisense, but still fails to build giving error "Visual Basic 10.0 does not support async methods...".  

BoxConfig boxConfig = new BoxConfig(CLIENT_ID, CLIENT_SECRET, new Uri("https://xxxxxxxx.com"));

BoxClient client = new BoxClient(boxConfig);

Dim boxItems As BoxCollection(Of Models.BoxItem) = Await boxClient.FoldersManager.GetFolderItemsAsync("0", 500)

I've got VS2012 update 4, and looking at looking at VS help, have Visual Basic 2012.
